src/base/random.cc

```diff
@@ -42,6 +42,8 @@
 
 #include <sstream>
 
+#include <algorithm>
```

Comment:
I think this should go above `sstream` (in alphabetical order) and shouldn't be separated by spaces.